packages/git-client/src/ConventionalGitClient.ts

```diff
@@ -135,7 +135,11 @@ export class ConventionalGitClient extends GitClient {
    * @returns Last semver tag, `null` if not found.
    */
   async getLastSemverTag(params: GetSemverTagsParams & Params = {}) {
-    return (await this.getSemverTags(params).next()).value || null
+    const semverTagsStream = this.getSemverTags(params)
+    const lastSemverTag = (await semverTagsStream.next()).value || null
+
+    await semverTagsStream.return()
+    return lastSemverTag
```

Comment:
```suggestion
    const stream = this.getSemverTags(params)
   
    for await (const tag of stream) {
      return tag
    }
   
    return null
```

I think it more readable

Reply:
I agree with you.
I also was not happy with the readability of the previous solution.

I extracted the code in a function to avoid duplication and also have only one `// eslint-disable-next-line no-unreachable-loop`.
I've added tests for it.

I added a test for `getLastTag` and `getLastSemverTags`, not sure they are very useful.
Let me know if you want to keep them or not.